src/praisonai/tests/unit/bots/test_delivery_control_store.py

```
@@ -0,0 +1,282 @@
+        reg.mark_dead("p", "3", reason="r")
+        assert reg.size() == 2
+        assert reg.is_dead("p", "1") is False
+        assert reg.is_dead("p", "3") is True
+
+    def test_ttl_expiry(self, tmp_path):
+        store = DeliveryControlStore(tmp_path / "ctl.sqlite")
+        reg = DeadTargetRegistry(store=store, ttl_seconds=1)
+        reg.mark_dead("telegram", "-1001", reason="403")
+        # Force ts into the past directly in the store.
+        store.dead_clear("telegram", "-1001")
+        store._lock  # sanity: store exists
```

Ruff (0.15.20)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)